router/h2/src/main/scala/io/buoyant/router/h2/ClassifiedRetryFilter.scala

```diff
@@ -117,6 +117,8 @@ class ClassifiedRetryFilter(
 
         // Attempt early classification before the stream is complete.
         responseClassifier(H2ReqRep(req, Return(rsp))) match {
+          case Some(ResponseClass.Ignorable | ResponseClass.Failed(false)) =>
```

Comment:
`ResponseClass.Failed(false)` is redundantly handled in this and the next branch